modules/gateway-task/config.tf

```diff
@@ -0,0 +1,43 @@
+locals {
+  consulLogin = var.acls ? {
+    enabled = var.acls
+    method  = var.service_token_auth_method_name
+    // TODO: Move this to a top-level partition field in the CONSUL_ECS_CONFIG_JSON
+    extraLoginFlags = var.consul_partition != "" ? ["-partition", var.consul_partition] : []
+  } : null
+
+  consul_service_meta = var.consul_service_meta
+
+  // if mesh gateway WAN federation is enabled add the metadata to the gateway service registration that exposes the Consul servers.
+  consul_gateway_meta = var.enable_mesh_gateway_wan_federation ? { "consul-wan-federation" : "1" } : {}
+
+  config = {
+    consulHTTPAddr   = var.consul_http_addr
+    consulCACertFile = var.consul_https_ca_cert_arn != "" ? "/consul/consul-https-ca-cert.pem" : ""
+    consulLogin      = local.consulLogin
+    service = {
+      name      = local.service_name
+      tags      = var.consul_service_tags
+      port      = 0 // note: field is required by consul-ecs schema. Not used for gateway.
+      meta      = local.consul_service_meta
+      namespace = var.consul_namespace
+      partition = var.consul_partition
+    }
```

Comment:
Are we using this `service` block to configure these fields for the gateway service?

I was thinking we'd use only the `gateway` field to configure gateways. All of these `service.*` fields were duplicated on to the `gateway` object in the schema.json for that reason: https://github.com/hashicorp/consul-ecs/blob/4bf464a8df2645f1e28e1168ffa2f2bbcd2ef82a/config/schema.json#L349-L430

What do you think?